upup/pkg/fi/cloudup/azure/azure_cloud.go

```diff
@@ -128,9 +134,34 @@ func (c *azureCloudImplementation) AddClusterTags(tags map[string]*string) {
 }
 
 func (c *azureCloudImplementation) GetApiIngressStatus(cluster *kops.Cluster) ([]kops.ApiIngressStatus, error) {
-	// TODO(kenji): Implement this. Currently we return nil as we
-	// don't create any resources for ingress to the API server.
-	return nil, nil
+	var ingresses []kops.ApiIngressStatus
+
+	// Get scale sets in cluster resource group and find masters scale set
+	scaleSets, err := c.vmscaleSetsClient.List(context.TODO(), cluster.Spec.CloudConfig.Azure.ResourceGroupName)
+	if err != nil {
+		return nil, fmt.Errorf("error getting Cluster Master Scale Set for API Ingress Status: %s", err)
+	}
+	var vmssName string
+	for i := range scaleSets {
+		if strings.Contains(*scaleSets[i].Name, ".masters.") {
```

Comment:
Wondering if there is a better way to do.. Maybe using tags is better? I believe other people can better chime-in here.

Reply:
I agree tags would be better here. This is how openstack does it:

https://github.com/kubernetes/kops/blob/55e38c96d3b83c7dd7c89c205fb0a1f2914e7851/upup/pkg/fi/cloudup/openstack/cloud.go#L704-L709